src/main/java/us/abstracta/wiresham/VirtualTcpService.java

```diff
@@ -18,86 +16,33 @@
 
-  private int port = DYNAMIC_PORT;
-  private Flow flow;
-  private boolean sslEnabled;
-  private SSLContext sslContext;
-  private int readBufferSize = DEFAULT_READ_BUFFER_SIZE;
   private int maxConnections = DEFAULT_MAX_CONNECTION_COUNT;
   private boolean stopped = false;
-  private final Set<ConnectionFlowDriver> connectionDrivers = new HashSet<>();
   private ExecutorService serverExecutorService;
-  private ExecutorService clientExecutorService;
-  private ServerSocket server;
 
-  public int getPort() {
-    return server.getLocalPort();
-  }
-
-  public void setPort(int port) {
-    this.port = port;
-  }
-
-  public void setFlow(Flow flow) {
-    this.flow = flow;
-    Optional<PacketStep> bigPacketStep = flow.getSteps().stream()
-        .filter(s -> s instanceof ReceivePacketStep && s.data.getBytes().length > readBufferSize)
-        .findAny();
-    if (bigPacketStep.isPresent()) {
-      throw new IllegalArgumentException(String.format(
-          "Read buffer size of %d bytes is not enough for receiving expected packet from client "
-              + "with %s", readBufferSize, bigPacketStep.get().data));
-    }
-  }
-
-  /**
-   * @deprecated use {@link #setSslContext(SSLContext)} instead, potentially using {@link
-   * SSLContext#getDefault} as parameter.
-   */
-  @Deprecated
-  public void setSslEnabled(boolean sslEnabled) {
-    this.sslEnabled = sslEnabled;
-  }
-
-  public void setSslContext(SSLContext sslContext) {
-    this.sslContext = sslContext;
-  }
-
-  public void setReadBufferSize(int readBufferSize) {
-    this.readBufferSize = readBufferSize;
-  }
-
-  public void setMaxConnections(int maxConnections) {
-    this.maxConnections = maxConnections;
-  }
-
-  public void start() throws IOException {
+  public void start() {
     stopped = false;
     serverExecutorService = Executors.newSingleThreadExecutor();
-    clientExecutorService = Executors.newFixedThreadPool(maxConnections);
-    if (sslContext != null) {
-      server = sslContext.getServerSocketFactory().createServerSocket(port);
-    } else {
-      server = new ServerSocket(port);
-    }
+    virtualExecutorService = Executors.newFixedThreadPool(maxConnections);
     serverExecutorService.submit(this);
   }
 
   @Override
   public void run() {
-    LOG.debug("Starting server on {} with flow: {}", server.getLocalPort(), flow);
-    LOG.info("Waiting for connections on {}", server.getLocalPort());
     while (!stopped) {
       try {
-        addClient(new ConnectionFlowDriver(server.accept(), readBufferSize, flow));
+        for (List<PacketStep> steps : packetsByPortMap.keySet()) {
+          ServerSocket server = packetsByPortMap.get(steps);
+          LOG.debug("Starting server on {} with steps: {}", server.getLocalPort(), steps);
+          LOG.info("Waiting for connections on {}", server.getLocalPort());
+          addClient(new ConnectionFlowStepsDriver(server.accept(), readBufferSize, steps));
```

Comment:
if a flow has usage of one port, then another, then back to the first one, are you creating two server ports?